main.go

```diff
@@ -269,12 +329,65 @@ func _main() error {
 		return err
 	}
 	defer os.RemoveAll(tmpDir)
+	if cache {
+		tmpDir = "dsq-cache"
+	}
+
+	settings := *runner.DefaultSettings
+	ec := runner.NewEvalContext(settings, tmpDir)
+
+	if cache {
+		// sqlite file is present on disk
```

Comment:
I would move all of this into a separate function that returns an error.

If the error is not nil we should not fail we should fall back to not using the cached version. So it should just log the error and keep going

Reply:
After my other comments I think you just don't need this section at all.

But still, if anything fails in cache mode it shouldn't fail the whole command it should just log and fall back to the non-cached version.